src/components/CommentForm/CommentForm.tsx

```diff
@@ -233,8 +233,32 @@ export const CommentForm = ({
     }
   };
 
-  if (firstPost) {
-    return <FirstCommentWelcome />;
+  const submitUserName = async (userName: string) => {
+    setError("");
+    if (!userName) {
+      setError("Username field cannot be empty");
+      return;
+    }
+
+    const response = await addUserName(userName);
+    if (response.status === "ok") {
+      // If we are able to submit userName we should continue with submitting comment
+      submitForm();
+      setFirstPost(false);
+    } else {
+      response.errors && setError(response.errors[0].message);
+    }
+  };
+
+  if (firstPost && body) {
```

Comment:
Can we rename `firstPost` to `usernameMissing`? I think that will read a lot better